homeassistant/components/proximity/__init__.py

```diff
@@ -203,16 +205,21 @@ def __init__(
         self._attr_unit_of_measurement = self.coordinator.unit_of_measurement
 
     @property
-    def state(self) -> str | int | float:
+    def data(self) -> dict[str, str | int | None]:
+        """Get data from coordinator."""
+        return self.coordinator.data.proximity
+
+    @property
+    def state(self) -> str | float:
         """Return the state."""
-        return self.coordinator.data.proximity[ATTR_DIST_TO]
+        if isinstance(distance := self.data[ATTR_DIST_TO], str):
+            return distance
+        return self.coordinator.convert_legacy(cast(int, distance))
 
     @property
     def extra_state_attributes(self) -> dict[str, str]:
         """Return the state attributes."""
         return {
-            ATTR_DIR_OF_TRAVEL: str(
-                self.coordinator.data.proximity[ATTR_DIR_OF_TRAVEL]
-            ),
-            ATTR_NEAREST: str(self.coordinator.data.proximity[ATTR_NEAREST]),
+            ATTR_DIR_OF_TRAVEL: str(self.data[ATTR_DIR_OF_TRAVEL] or STATE_UNKNOWN),
```

Comment:
We use `None` for unknown values.

Reply:
I don't think we should keep legacy behavior around. We are already making big changes now, now it the time I would say.

Reply:
the whole proximity entity will be removed with 2024.8 and users will be informed to change away from the legacy proximity entity. further changing this behavior would cause another breaking change, where the user is forced to change existing automations/scripts instant after update to 2024.2, instead of make use of the 6 months deprecation period.

Reply:
just to make it clear - the new sensor entities properly set `None` if needed

Reply:
👍 Ok right, yeah that makes sense 👍